sky/spot/controller.py

```diff
@@ -29,10 +29,18 @@ def __init__(self, job_id: int, task_yaml: str,
         self._job_id = job_id
         self._task_name = pathlib.Path(task_yaml).stem
         self._task = sky.Task.from_yaml(task_yaml)
+
         self._retry_until_up = retry_until_up
         # TODO(zhwu): this assumes the specific backend.
         self.backend = cloud_vm_ray_backend.CloudVmRayBackend()
 
+        # Add a unique identifier to the task environment variables, so that
+        # the user can have the same id for multiple recoveries.
+        task_envs = self._task.envs or {}
+        task_envs['SKYPILOT_SPOT_RUN_ID'] = (f'{self.backend.run_timestamp}'
+                                             f'_id-{self._job_id}')
```

Comment:
nit: show an example value? Maybe change `_id-` -> `_` to make it shorter?

Reply:
An example value would be: `sky-2022-10-04-22-46-52-467694_id-17`. I feel like withhout the `_id`, it will be a bit confusing what the last number is at the end. Wdyt?

Reply:
Done. Thanks!